addons/pos_loyalty/static/src/overrides/models/pos_order_line.js

```diff
@@ -42,6 +42,13 @@ patch(PosOrderline, {
 });
 
 patch(PosOrderline.prototype, {
+    setup(vals) {
+        if (this.is_reward_line && !this.reward_id) {
+            this.delete();
```

Comment:
Deleting an item from itself seems a bit weird. Maybe it can be done at the same place as `couponPointChanges`. 

WDYT @adgu-odoo  ?